### ISSUE TITLE
Add Clone method to the classes that will create a clone of just the properties (excluding navigation properties)

### ISSUE THREAD

**Issue by NaomiN** [ReversePOCO/EntityFramework-Reverse-POCO-Code-First-Generator#445]
I am wondering if it would be possible to add a new 'Clone()' method that will create a copy of the current instance of the class but as a new object where only 'simple' properties will be taken from the current instance and the PK property will be empty? I had a need for such method today and at the end I manually copied all the properties because MemberwiseClone() didn't do what I needed it to do (it copied navigation properties too).

**Comment by sjh37** [comment]
Take a look at AutoMapper

**Comment by NaomiN** [comment]
I know about AutoMapper, but I'm not sure if it creates clone of just simple properties or would have copied navigation properties as well if I would create a map to itself.

In any case, for the particular scenario where I was thinking about a clone I eventually just set all navigation properties to null in the original object and that solved the issue I was having (I was trying to create a duplicate of the purchase order with line items).

**Comment by sjh37** [comment]
I think AutoMapper will do shallow or deep clones. Just depends on how you configure it.

**Comment by NaomiN** [comment]
Thanks, Simon, I may look into AutoMapper in more depth (I did post a comment yesterday in Stackoverflow but don't think I got any response). For the particular scenario I wanted that I was able to solve the problem without AutoMapper by just setting all navigation properties to null. I think still some generic 'Clone' method may be helpful and it's probably easy to implement while generating the object. E.g. maybe someone will find that suggestion useful.
